firmware/Core/Src/telecommands/uart_telelecommand_defs.c

```diff
@@ -0,0 +1,313 @@
+
+    LOG_system_enum_t LOG_source = LOG_SYSTEM_TELECOMMAND;
+    
+    // UART 1 selected (MPI)
+    if(strcasecmp(arg_uart_port_name, "MPI") == 0) {
+
+        // Set log source
+        LOG_source = LOG_SYSTEM_MPI;
+        UART_rx_buffer_size_ptr = &UART_mpi_buffer_len;
+
+        // Transmit bytes and receive response (DMA enabled reception)
+        const uint8_t MPI_tx_rx_status = MPI_send_telecommand_get_response(tx_buffer, tx_buffer_len, rx_buffer, *UART_rx_buffer_size_ptr, &rx_buffer_len);
+
+        // Log successful transmission and print rx_buffer response in hex (uint8_t)
+        char rx_buffer_str[rx_buffer_len*3];
+
+        for(uint16_t i = 0; i < rx_buffer_len; i++) {
+            // Copy received bytes into a string to log it
+            sprintf(&rx_buffer_str[i*3], "%02X ", rx_buffer[i]);
+        }
+
+        // Log mpi response
+        switch(MPI_tx_rx_status) {
+            case 0:
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_NORMAL, LOG_SINK_ALL,
+                    "Transmitted %u byte(s) to %s.", 
+                    tx_buffer_len, 
+                    arg_uart_port_name
+                );
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_NORMAL, LOG_SINK_ALL,
+                    "Received %u byte(s) from %s: %s",
+                    rx_buffer_len,
+                    arg_uart_port_name,
+                    rx_buffer_str
+                );
+                return 0;   // Success
+            case 2:
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                    "Failed UART transmission."
+                );
+                return 3;   // Error code: Error transmitting data
+            case 3:
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                    "Failed UART reception."
+                );
+                return 4;   // Error code: Error receiving data
+            case 4:
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                    "No response received. Timeout waiting for 1st byte."
+                ); 
+                return 5;   // Error code: Error receiving data
+            case 8:
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                    "Not enough space in the MPI response buffer"
+                ); 
+                return 6;   // Error code: Error receiving data
+            default:
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                    "Unknown response return: %u", 
+                    MPI_tx_rx_status
+                );
+                return 7;   // Error code: Unhandled error
+        }
+    }
+
+    // UART 4 Selected (CAMERA)
+    else if(strcasecmp(arg_uart_port_name, "CAMERA") == 0) {
```

Comment:
You need to return and error and say that this is not implemented. Currently it just fails silently.

Some electrical team members wasted a lot of time trying to troubleshoot this UART port tonight, despite not being implemented.

Why isn't this implemented?

Reply:
For the camera since it utilizes DMA very much like the MPI, I had planned to implement it the same way as the MPI with a modularized send/receive function once the Camera code was added to main. My main reasoning for this was to avoid multiple implementations of the same send/receive functionality.

My bad on missing the error message earlier—it’s definitely crucial here.

I can add the basic TX functionality here if that is preferred, I was just a bit optimistic seeing the progress on the Camera code that I decided to just wait on its push before adding it.